src/ofonotracker.cpp

```diff
@@ -0,0 +1,335 @@
+ * OFonoTrackerPrivate
+ * ========================================================================= */
+
+class OFonoTrackerPrivate : public QObject, protected QDBusContext
+{
+    Q_OBJECT;
+public:
+    OFonoTrackerPrivate(bool synchronousInit, OFonoTracker *parent);
+    virtual ~OFonoTrackerPrivate();
+    QStringList imeiNumbers() const;
+private Q_SLOTS:
+    void evaluateProperties();
+    void handleManagerModemAdded(QDBusObjectPath objectPath, QVariantMap properties);
+    void handleManagerModemRemoved(QDBusObjectPath objectPath);
+    void handleModemPropertyChanged(QString key, QDBusVariant value);
+private:
+    void evaluatePropertiesLater();
+    void getModemsAsync();
+    void getModemsSync();
+    void flushModems();
+    void setModems(const ModemInfoList &modemInfoList);
+    void addModem(const ModemInfo &modemInfo);
+    void removeModem(QString path);
+    void setModemProperty(QString path, QString key, QVariant val);
+    static bool trackedModemProperty(const QString &key);
+    OFonoTracker *q_ptr;
+    QDBusConnection m_systemBus;
+    QStringList m_modemPathList;
+    QHash<QString, ModemInfo> m_modemInfoHash;
+    QTimer *m_evaluatePropertiesTimer;
+    QStringList m_imeiNumbers;
+    Q_DISABLE_COPY(OFonoTrackerPrivate);
+    Q_DECLARE_PUBLIC(OFonoTracker);
+};
+
+OFonoTrackerPrivate::OFonoTrackerPrivate(bool synchronousInit, OFonoTracker *parent)
+    : QObject(parent)
+    , q_ptr(parent)
+    , m_systemBus(QDBusConnection::systemBus())
+    , m_evaluatePropertiesTimer(new QTimer(this))
+{
+    registerDBusTypes();
+    m_evaluatePropertiesTimer->setSingleShot(true);
+    m_evaluatePropertiesTimer->setInterval(20);
+    connect(m_evaluatePropertiesTimer, &QTimer::timeout, this, &OFonoTrackerPrivate::evaluateProperties);
+    m_systemBus.connect(OFONO_SERVICE, OFONO_MANAGER_OBJECT, OFONO_MANAGER_INTERFACE, OFONO_MANAGER_MODEM_ADDED,
+                        this, SLOT(handleManagerModemAdded(QDBusObjectPath, QVariantMap)));
+    m_systemBus.connect(OFONO_SERVICE, OFONO_MANAGER_OBJECT, OFONO_MANAGER_INTERFACE, OFONO_MANAGER_MODEM_REMOVED,
+                        this, SLOT(handleManagerModemRemoved(QDBusObjectPath)));
+    m_systemBus.connect(OFONO_SERVICE, QString(), OFONO_MODEM_INTERFACE, OFONO_MODEM_PROPERTY_CHANGED,
+                        this, SLOT(handleModemPropertyChanged(QString, QDBusVariant)));
+    /* For the benefit of existing code that assumes also D-Bus based values
+     * to be available immediately after the object is instantiated -> make
+     * it possible to utilize a synchronous probe and evaluate property values
+     * before returning from constructor. */
+    if (synchronousInit)
+        getModemsSync();
+    else
+        getModemsAsync();
+}
+
+OFonoTrackerPrivate::~OFonoTrackerPrivate()
+{
+}
+
+QStringList OFonoTrackerPrivate::imeiNumbers() const
+{
+    qWarning() << "#### call imeiNumbers() ->" << m_imeiNumbers;
```

Comment:
Looks like debug logging. I recommend that you add a logging category for those that you intend to keep and use `qCDebug`.

Reply:
The intent/need to remove these was expressed in bz, but I'll just cut them out now.

Reply:
I commented this first and only after that read the message. Yes, it's all good.